xarray/core/rolling.py

```diff
@@ -254,45 +275,41 @@ def wrapped_func(self, **kwargs):
 
             axis = self.obj.get_axis_num(self.dim)
 
-            if isinstance(self.obj.data, dask_array_type):
+            padded = self.obj.variable
+            if self.center:
+                shift = (-self.window // 2) + 1
+
+                if (LooseVersion(np.__version__) < LooseVersion('1.13') and
+                        self.obj.dtype.kind == 'b'):
+                    # with numpy < 1.13 bottleneck cannot handle np.nan-Boolean
+                    # mixed array correctly. We cast boolean array to float.
```

Comment:
I notice that with the older numpy and benchmark, reducing `np.nan`-and-boolean-mixed arrays gives a wrong result.
Here is a workaround, where we cast boolean arrays to float.